File: source/manual/ds_writing/writing_java/StepperMotorClass4.java
public void attribute_factory(Vector att) throws DevFailed
{
	att.addElement(new Attr("Position",
				Tango_DEV_LONG,
				AttrWriteType.READ_WITH_WRITE,
				"SetPosition"));

	UserDefaultAttrProp def_prop = new UserDefaultAttrProp();
	def_prop.set_label("set the motor position");
	def_prop.set_format("scientific;setprecision(4)");
	Attr at = new Attr("SetPosition",
			   Tango_DEV_LONG,
			   AttrWriteType.WRITE));
	at.set_default_properties(def_prop);
	att.addElement(at);

	att.addElement(new Attr("Direction",
				Tango_DEV_LONG));
}

File: source/manual/t_b_source/ds_writing/writing/steppermotorclass4.cpp
void StepperMotorClass::attribute_factory(vector<Tango::Attr *> &att_list)
{
	att_list.push_back(new PositionAttr());

	Tango::UserDefaultAttrProp def_prop;
	def_prop.set_label("Set the motor position");
	def_prop.set_format("scientific;setprecision(4)");
	Tango::Attr *at = new SetPositionAttr();
	at->set_default_properties(def_prop);
	att_list.push_back(at);

	att_list.push_back(new DirectcionAttr());
}
